### ISSUE TITLE
Feature request: per-input compression

### ISSUE THREAD

**Issue by KryptonicDragon** [mateoconlechuga/convbin#8]
According to the info that convbin outputs when you run it without arguments, it says
> -i, --input \<file\> Input file. Can be specified multiple times, input files are appended in order.

I find the ability to append multiple input files very useful, but one problem I have run into is that `--compress` compresses the entire output.
This makes it impossible to, in one run of convbin, append together data that I want compressed and data that I don't want compressed. If there was another option (or if the current one was modified to behave this way, but that would probably be problematic and break things) that sets compression per-input (like how each `--iformat` only applies to the following `--input`).

TL;DR: A `--compress`-like option that applies to only one `--input` when using multiple inputs in one command.

**Comment by adriweb** [comment]
I'm not sure that makes sense, because we're having only one output regardless of how many inputs there are; right? In which case, the option is designed to control whether that output is compressed or not: it's not related to the input(s).

(The special case of having a large program being split in a launcher+appvars is an exception I guess but not what you seem to be talking about anyway)

**Comment by KryptonicDragon** [comment]
The way I see it, having only the option to enable or disable compression on the entire output forces you to treat all the data in the output as one thing. If you've opted for compression, you're left with one atomic "blob" that you're unable to access smaller parts of, without having to allocate a buffer and decompress the entire thing first. If your use-case is to store multiple _different_ things in one .8xv, that memory cost can be a hurdle.
Having to treat the entire output as one atomic thing (if you used compression) seems like it decreases the usefulness of the ability to concatenate multiple non-atomic things together (using multiple `--input`s), no?

However, if there was another option that enabled compression per-input, it would allow for the use-case of putting data that you don't ever need to access all-at-once (I.E my particular use case that led me to requesting this feature). Then you only ever need as much memory (for holding the decompressed data) as the data you're actively intending to use.

Although, it does depend on what your output format is. In my particular use case, I'm outputting to an .8xv. It's not a self-extracting executable or anything like that, so there's no reason that I can't or wouldn't want _only some parts_ of the final output to be compressed.

Options for working around the lack of such a feature are:
-Don't compress anything, no space savings.
-Compress the entire thing, and hope you have enough bss and/or heap to handle hold all the decompressed data for the entire duration that any of it may be required.
-Run convbin multiple times, in two phases. For the first phase it's nothing more than a compression tool; you run it once for each chunk of data, outputting to a simple .bin file for each, and choosing which ones to compress. Then you run convbin once more, using it as a "format conversion" tool, concatenating all the outputs from the first phase into your final output, I.E an .8xv.

**Comment by mateoconlechuga** [comment]
Dude have you ever heard of a script.

**Comment by adriweb** [comment]
For future reference, see #9 .
